routing/pathfind.go

```diff
@@ -99,11 +99,12 @@ type edgePolicyWithSource struct {
 // finalHopParams encapsulates various parameters for route construction that
 // apply to the final hop in a route. These features include basic payment data
 // such as amounts and cltvs, as well as more complex features like destination
-// TLV records.
+// TLV records and payment address.
 type finalHopParams struct {
-	amt       lnwire.MilliSatoshi
-	cltvDelta uint16
-	records   record.CustomSet
+	amt         lnwire.MilliSatoshi
+	cltvDelta   uint16
+	records     record.CustomSet
+	paymentAddr *[32]byte
```

Comment:
Can this struct `finalHopParams` be reused to pass into findPath?

Reply:
do we gain anything functionally by doing so? in this pr, we would only be using 2 of the 4 fields, perhaps it is  better left to when `findPath` and `newRoute` are fully unified? given how long this pr has been open, i would prefer to defer pure refactorings so that we can unblock other changes

Reply:
Discussed offline, future pr.